drizzlepac/haputils/analyze.py

```diff
@@ -425,13 +426,32 @@ def analyze_data(input_file_list, log_level=logutil.logging.DEBUG, type=""):
         # Determine if the image has one of these conditions.  The routine
         # will exit processing upon the first satisfied condition.
 
+        # Check if all science image arrays in the RAW file are filled with zero values
+        non_zero_data_in_array = False # start assuming data is zeros
+        science_ext_ind_array = count_sci_extensions(input_file, return_ind=True)
+        # make sure science extension exists
+        if len(science_ext_ind_array)>0:
+            for sci_ext_ind in science_ext_ind_array:
+                science_data = getdata(input_file, sci_ext_ind)
+                # change flag if good data in any science extensnion array
```

Comment:
spelling: extension (Line 436)

Reply:
Fixed. I clearly have an issue with typing the word extension.